### PR TITLE
fix wild cards handling

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_aad.py [hunk 1]
```diff
@@ -389,15 +392,15 @@ def test_list_revisions_key_label(self):
         assert cnt >= 2
 
     def test_list_revisions_only_label(self):
-        items = self.app_config_client.list_revisions(labels=[LABEL])
+        items = self.app_config_client.list_revisions(labels=LABEL)
         cnt = 0
         for kv in items:
             assert kv.label == LABEL
             cnt += 1
         assert cnt >= 1
```
Comment (bryevdv) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_aad.py]: All your changes LGTM (I'm assuming there are corresponding swagger changes for the models args?) My only comment is noticing the following test pattern which I think could be expressed more clearly/concisely if you are felt like updating them:
```
items = list(self.app_config_client.list_revisions(labels=LABEL))
assert len(items) >= 1
assert all(x.label == LABEL for x in items)
```

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py [hunk 2]
```diff
@@ -358,7 +337,7 @@ def test_list_configuration_settings_multi_pages(self):
             pass
 
     def test_list_configuration_settings_null_label(self):
-        items = self.app_config_client.list_configuration_settings(labels=[""])
+        items = self.app_config_client.list_configuration_settings(label_filter="\0")
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py]: Doesn't seem right, you used to have code code to replace an empty string with "\0" so customer never see "\0", why not keeping it?

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py]: Yes. We used to escape the char for the user and in the new design, we will let user escape it by themselves. This is a change by design.

**Pull request merged**